linux/guest_customization/linux_gosc_start.yml

```diff
@@ -41,9 +41,42 @@
     type: "{{ ip_assignment }}"
     timeout: 900
 
-# Wait for GOSC completed
 - include_tasks: ../../common/vm_wait_gosc_completed.yml
 
+# Wait for GOS is rebooted by checking tools status
+- block:
```

Comment:
I think this is not a good way to say VM is reboot or not just according to the VMware tools status got at that time. What if VM reboots and tools status is running at the time getting tools status, what if there is something wrong with tools service after GOSC and it is not running at the time getting tools status. It's hard to say VM is rebooting, rebooted, or not.

If check other configurations is enough here for now, we can consider to add this check later when there is better solution found.